torch/csrc/jit/ir/ir.cpp

```diff
@@ -1608,17 +1608,30 @@ Node* Graph::createTupleIndex(
   return n;
 }
 
-Node* Graph::createTupleSlice(Value* tup, int64_t beg, int64_t end) {
-  auto n = create(prim::TupleSlice, {tup});
-  auto tuple_type = tup->type()->expect<TupleType>();
-  n->i_(attr::beg, beg);
-  n->i_(attr::end, end);
-  std::vector<TypePtr> output_types;
-  for (auto i = beg; i < end; ++i) {
-    output_types.push_back(tuple_type->elements().at(i));
-  }
-  auto tt = TupleType::create(std::move(output_types));
-  n->output()->setType(tt);
+Node* Graph::createTupleSlice(
+    Value* tup,
+    int64_t beg,
+    int64_t end,
+    int64_t step_size) {
+  std::vector<Value*> newValues;
+  TupleTypePtr tt = tup->type()->expect<TupleType>();
+
+  if (step_size > 0) {
```

Comment:
Now that I think about it. It is a bit confusing if we flip `begin` and `end` earlier but now always iterate from begin and end, but with a different loop condition `i > end` vs `i < end` here. IMHO, this logic would be more readable if we put all logic about checking step-size being positive or negative then perform different actions here. Let me know what you think.

Reply:
yeah that is a good idea. I will try to move the logic into one step